src/org/labkey/test/tests/SampleTypeNameExpressionTest.java

```diff
@@ -176,6 +180,105 @@ public void testSimpleNameExpression()
+
+        CreateSampleTypePage createPage = sampleTypeHelper.goToCreateNewSampleType();
+
+        createPage.setName(sampleTypeName);
+
+        String tricky01 = "Søµ∑,String,,,";
+        String tricky02 = "NE∫,";
+        String tricky03 = "@-\\*-";
+        String tricky04 = "+{My'Text}=%#$"; // These curly braces cause the warning when saving.
+        String tricky05 = String.format("@\"%s", tricky01);
+        String dateFormat = "yy-MM-dd";
+        int counterStart = 500;
+
+        //Søµ∑,String,,,${NE∫,:withCounter(500)}@-\\*-${genId:number('000,000')}+{My'Text}=%#$${now:date('yy-MM-dd')}@\"Søµ∑,String,,,
+        String nameExpression = String.format("%s${%s:withCounter(%d)}%s${genId:number('000,000')}%s${now:date('%s')}%s",
+                tricky01, tricky02, counterStart, tricky03, tricky04, dateFormat, tricky05);
+
+        log(String.format("Use name expression: '%s'.", nameExpression));
+
+        createPage.setNameExpression(nameExpression);
+
+        log("Click the 'Save' button and wait for the warning dialog.");
+        Locator.button("Save").findElement(getDriver()).click();
+
+        ModalDialog dialog = null;
+
+        try
+        {
+            dialog = new ModalDialog.ModalDialogFinder(getDriver()).withTitle("Naming Pattern Warning(s)").waitFor();
+        }
+        catch (TimeoutException toe)
+        {
+            checker().withScreenshot("No_Warning_Dialog")
+                    .error("The expected warning dialog for this name expression did not show up.");
+        }
+
+        if(null != dialog)
+        {
+            log("Verify that you can save with the warning.");
+            dialog.dismiss("Save anyways...", 2_500);
+        }
+
+        log("Validate save worked. Should be sent back to the project begin page and there should be a link with the sample type name.");
+
+        checker().fatal()
+                .verifyTrue("Did not successfully save the sample type. Fatal error.",
+                        waitFor(()->Locator.linkWithText(sampleTypeName).findWhenNeeded(getDriver()).isDisplayed(), 1_000));
+
+        log("Create some samples and validate the name(s).");
+
+        int samplesCount = 4;
+        String pasteData = """
+                Description
+                A
+                B
+                C
+                D""";
+
+        clickAndWait(Locator.linkWithText(sampleTypeName));
+        var dataRegion = DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+        var importDataPage = dataRegion.clickImportBulkData();
+        importDataPage.setCopyPasteInsertOption(false);
+        importDataPage.selectCopyPaste()
+                .setFormat(ImportDataPage.Format.TSV)
+                .setText(pasteData)
+                .submit();
+
+        // Get the date as soon after sample creation as possible.
+        LocalDateTime ldt = LocalDateTime.now();
+        String date = DateTimeFormatter.ofPattern(dateFormat, Locale.ENGLISH).format(ldt);
```

Comment:
I predict some random failures at some point in the future when the test spans the midnight hour, but I hope that should be pretty rare.

Reply:
Yep, and I had thought about removing this from the test. However, there are no test that uses the 'now' option in the name expression, so I thought the coverage would be good. 
If/when it does fail because of that hopefully it will be obvious.
If it happens a lot, I'll remove it.